internals/setup.js

```diff
@@ -0,0 +1,397 @@
+#!/usr/bin/env node
+/* eslint-disable import/no-extraneous-dependencies */
+
+const readPkg = require('read-pkg')
+const writePkg = require('write-pkg')
+const shell = require('shelljs')
+const { exec } = require('child_process')
+const inquirer = require('inquirer')
+const ora = require('ora')
+const rimraf = require('rimraf')
+const compareVersions = require('compare-versions')
+const chalk = require('chalk')
+
+const npmConfig = require('./helpers/get-npm-config')
+
+process.stdin.resume()
+process.stdin.setEncoding('utf8')
+
+process.stdout.write('\n')
+
+/**
+ * Deletes the current directory
+ *
+ * @returns {Promise<any>}
+ */
+function deleteCurrentDir() {
+  return new Promise((resolve, reject) => {
+    rimraf(__dirname, error => {
+      if (error) {
+        reject(new Error(error))
+      } else {
+        resolve()
+      }
+    })
+  })
+}
+
+/**
+ * Checks if we are under Git version control
+ * @returns {Promise<boolean>}
+ */
+function hasGitRepository() {
+  return new Promise((resolve, reject) => {
+    exec('git status', (err, stdout) => {
+      if (err) {
+        reject(new Error(err))
+      }
+
+      const regex = new RegExp(/fatal:\s+Not\s+a\s+git\s+repository/, 'i')
+
+      /* eslint-disable-next-line no-unused-expressions */
+      regex.test(stdout) ? resolve(false) : resolve(true)
+    })
+  })
+}
+
+/**
+ * Checks if this is a clone from our repo
+ * @returns {Promise<any>}
+ */
+function checkIfRepositoryIsAClone() {
+  return new Promise((resolve, reject) => {
+    exec('git remote -v', (err, stdout) => {
+      if (err) {
+        reject(new Error(err))
+      }
+
+      const isClonedRepo = stdout
+        .split(/\r?\n/)
+        .map(line => line.trim())
+        .filter(line => line.startsWith('origin'))
+        .filter(line => /kefranabg\/bento-starter\.git/.test(line)).length
+
+      resolve(!!isClonedRepo)
+    })
+  })
+}
+
+/**
+ * Remove the current Git repository
+ * @returns {Promise<any>}
+ */
+function removeGitRepository() {
+  return new Promise((resolve, reject) => {
+    try {
+      shell.rm('-rf', '.git/')
+      resolve()
+    } catch (err) {
+      reject(err)
+    }
+  })
+}
+
+/**
+ * Ask user if he wants to start with a new repository
+ * @returns {Promise<boolean>}
+ */
+function askUserIfWeShouldRemoveRepo() {
+  const NEW_REPOSITORY = 'NEW_REPOSITORY'
+  return inquirer
+    .prompt([
+      {
+        type: 'confirm',
+        message: 'Do you want to start with a new repository ?',
+        name: NEW_REPOSITORY
+      }
+    ])
+    .then(answers => !!answers[NEW_REPOSITORY])
+}
+
+/**
+ * Ask user for new origin for this repository. If provided, call git remote add origin
+ * @returns {Promise<any>}
+ */
+const askUserForNewRemote = () => {
+  const NEW_REMOTE = 'NEW_REMOTE'
+
+  return inquirer
+    .prompt([
+      {
+        type: 'input',
+        message: 'Enter new remote for this repository [Enter to cancel]: ',
```

Comment:
I think it could be great to add an example of origin url : 
```suggestion
        message: 'Enter new remote origin (ex: https://github.com/user/my-repo.git) for this repository [Enter to cancel]:  ',
```